### PR TITLE
[CLC-312]: Add "cancel" command to DMT

### PR DESCRIPTION
**Pull request by kutluhanmetin**

### CODE REVIEW

File: base/commands/migration/cancel_stages.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+func (st *CancelStages) cancelStage(ctx context.Context) func(stage.Statuser) error {
+	return func(statuser stage.Statuser) error {
+		c := CancelItem{ID: st.migrationID}
+		b, err := json.Marshal(c)
+		if err != nil {
+			return err
+		}
+		st.cancelQueue.Put(ctx, serialization.JSON(b))
+		return err
```
Comment (srknzl) [inline base/commands/migration/cancel_stages.go:80]:
```suggestion
		return nil
```

**Comment by srknzl** [comment]
@kutluhanmetin Note: We will revert back to IQueue usage instead of ITopic for cancel queue. FYI. So you should update PR as well. See https://github.com/hazelcast/hazelcast-enterprise/pull/6451#discussion_r1319510673

File: base/commands/migration/cancel_stages.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+func waitForCancel(ctx context.Context, ci *hazelcast.ClientInternal, migrationID string) error {
+	for {
+		s, err := fetchMigrationStatus(ctx, ci, migrationID)
+		if err != nil {
+			return err
+		}
+		if Status(s) == StatusCanceling {
+			return nil
+		}
```
Comment (srknzl) [inline base/commands/migration/cancel_stages.go:100]: any sleep here?

Comment (kutluhanmetin) [reply base/commands/migration/cancel_stages.go:100]: I remember we had a similar discussion for `WaitForMigrationToBeInProgress` and decided not to sleep. I think it is a similar situation in here as well.

**Review by srknzl — APPROVED** [review]

**Review by yuce — APPROVED** [review]

**Pull request merged**